pkg/registration/hub/manager.go

```diff
@@ -147,12 +155,32 @@ func (m *HubManagerOptions) RunControllerManagerWithInformers(
 	workInformers workv1informers.SharedInformerFactory,
 	addOnInformers addoninformers.SharedInformerFactory,
 ) error {
+
 	csrApprover, err := csr.NewCSRApprover(kubeClient, kubeInformers, m.ClusterAutoApprovalUsers, controllerContext.EventRecorder)
 	if err != nil {
 		return err
 	}
+	awsIRSAHubDriver, err := awsirsa.NewAWSIRSAHubDriver(ctx, m.HubClusterArn)
```

Comment:
this is a bit completed here. I was thinking an better approach would be

```go
var approvers []register.Approver
var drivers []register.HubDriver

for _, driverName := range m.EnabledRegistrationDrivers:
   switch driverName {
   case "csr":
       approvers = append(approvers, csr.NewCSRApprover(kubeClient, kubeInformers, m.ClusterAutoApprovalUsers, controllerContext.EventRecorder))
       drivers = append(drivers, csr.NewCSRHubDriver())     
   case "isra":
       ....  
   }
approver = register.NewAggregatedApprover(approvers...)
hubDriver = register.NewAggregatedHubDriver(drivers...)
```

Reply:
Addressed